packages/react-flight-dom-webpack/src/ReactFlightClientWebpackBundlerConfig.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export type ModuleMetaData = {
+  id: string,
+  chunks: Array<string>,
+};
+
+type Thenable = {
+  then(resolve: () => mixed, reject: (mixed) => mixed): mixed,
+  ...
+};
+
+// The chunk cache contains all the chunks we've preloaded so far.
+// If they're still pending they're a thenable. This map also exists
+// in Webpack but unfortunately it's not exposed so we have to
+// replicate it in user space. null means that it has already loaded.
+const chunkCache: Map<string, null | Thenable> = new Map();
+
+// Returning null means that all dependencies are fulfilled and we
+// can synchronously require the module now. A thenable is returned
+// that when resolved, means we can try again.
+export function preloadModule(moduleData: ModuleMetaData): null | Thenable {
+  let moduleEntry = require.cache[moduleData.id];
```

Comment:
Technically using `require.cache` and ESM is kind of illegal. It works in webpack in "automatic" JS mode (kind of mixed CJS and ESM), but `require.cache` is not available in strict ESM mode. At least it works for webpack <= 5.

Reply:
Have you considered exposing a Webpack specific helper to do something similar to this?

In this case I could just get rid of it since it’s just an optimization to avoid having to check many maps.

It might be nice to know for scheduling purposes though. E.g. we try to be very particular about when we call `__webpack_require__` since that can kick off a lot of synchronous work. So it’s nice to be able to know if the call will be free or hundreds of ms.